src/main/kotlin/org/jitsi/nlj/transform/node/incoming/IncomingStatisticsTracker.kt

```diff
@@ -219,25 +226,27 @@ class IncomingSsrcStats(
 
     /**
      * Notify this [IncomingSsrcStats] instance that an RTP packet [packet] for the stream it is
-     * tracking has been received and that it: was sent at [packetSentTimestampMs] (note this is NOT the
+     * tracking has been received and that it: was sent at [packetSentTimestamp] (note this is NOT the
      * raw RTP timestamp, but the 'translated' timestamp which is a function of the RTP timestamp and the clockrate)
      * and was received at [packetReceivedTimeMs]
      */
     fun packetReceived(
         packet: RtpPacket,
-        packetSentTimestampMs: Long,
-        packetReceivedTimeMs: Long
+        packetSentTimestamp: Instant,
+        packetReceivedTime: Instant
     ) {
         val packetSequenceNumber = packet.sequenceNumber
         synchronized(statsLock) {
             activitySinceLastSnapshot = true
             numReceivedPackets++
             numReceivedBytes += packet.length
-            val timeSincePreviousPacket = packetReceivedTimeMs - lastPacketReceivedTimeMs
-            if (timeSincePreviousPacket in 1..ACTIVITY_TIMEOUT_MS) {
-                durationActiveMs += timeSincePreviousPacket
+            if (lastPacketReceivedTime != null) {
+                val timeSincePreviousPacket = Duration.between(lastPacketReceivedTime, packetReceivedTime)
+                if (timeSincePreviousPacket in Duration.ZERO..ACTIVITY_TIMEOUT) {
```

Comment:
I think the previous logic (`if (timeSincePreviousPacket in 1..ACTIVITY_TIMEOUT_MS)`) was written that way so that we didn't add negative timestamps - though thinking about it, if we do get timestamp reordering, this will end up overestimating the durations (since we nonetheless update `lastPacketReceivedTime` in that case).

I was mostly trying to model the existing logic without working through the details too much.  Thinking about it, the simplification actually improves things.